packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -275,21 +275,58 @@ class DraggableScrollableController extends ChangeNotifier {
 ///
 ///   @override
 ///   Widget build(BuildContext context) {
+///     final ColorScheme colorScheme = Theme.of(context).colorScheme;
+///
 ///     return Scaffold(
 ///       appBar: AppBar(
 ///         title: const Text('DraggableScrollableSheet'),
 ///       ),
 ///       body: SizedBox.expand(
 ///         child: DraggableScrollableSheet(
 ///           builder: (BuildContext context, ScrollController scrollController) {
-///             return Container(
-///               color: Colors.blue[100],
-///               child: ListView.builder(
-///                 controller: scrollController,
-///                 itemCount: 25,
-///                 itemBuilder: (BuildContext context, int index) {
-///                   return ListTile(title: Text('Item $index'));
-///                 },
+///             return ColoredBox(
+///               color: colorScheme.primary,
+///               /// On Desktop and Web platforms, the vertical drag gesture is disabled by default
+///               /// to align with the natural behavior found in other desktop applications.
+///               /// However, if you still desire to make it work on these platforms,
+///               /// you can change the default scroll behavior to activate drag gestures by overriding dragDevices.
+///               /// For details, please refer to https://docs.flutter.dev/release/breaking-changes/default-scroll-behavior-drag
+///               child: ScrollConfiguration(
```

Comment:
We probably would want to use an actual drag gesture on the handle rather than enabling mouse drag scrolling like this. It can create issues when folks try to select text and it ends up scrolling the text instead. :(

Reply:
I see. I'm stuck with a trackpad issue. I'm trying to solve it if possible :) 

For context, I implemented an example as your suggestion by building a drag handle (please see sample code and demo vide as below). The idea is to get the delta y-axis from GestureDetector's onVerticalDragUpdate and then set that value to DraggableScrollableSheet's initialChildSize (as `sheetPosition` var)

It would be fine if the user held and dragged the handle only. But there is a side effect with trackpad when scrolling ListView. In case user scrolls ListView with trackpad, DraggableScrollableSheet will be expanded/collapsed along with its position but there is no suitable callback from DraggableScrollableSheet in order to update the corresponding `sheetPosition` (to sync with the drag handle). It causes the handle to not work afterward (you can see issue on demo video too).

Also, this issue is only happening with the **_trackpad_**, but it will also happen with **_mouse wheel_** as well once https://github.com/flutter/flutter/issues/101903 is implemented.

<details>
<summary>Sample code</summary>

```dart
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter/material.dart';

/// Flutter code sample for [DraggableScrollableSheet].

void main() => runApp(const DraggableScrollableSheetExampleApp());

class DraggableScrollableSheetExampleApp extends StatelessWidget {
  const DraggableScrollableSheetExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.blue.shade100),
      ),
      home: Scaffold(
        appBar: AppBar(
          title: const Text('DraggableScrollableSheet Sample'),
        ),
        body: const Example(),
      ),
    );
  }
}

class Example extends StatefulWidget {
  const Example({super.key});

  @override
  State<Example> createState() => _ExampleState();
}

class _ExampleState extends State<Example> {
  double sheetPosition = 0.5;
  final double dragSensitivity = 600;

  @override
  Widget build(BuildContext context) {
    final ColorScheme colorScheme = Theme.of(context).colorScheme;

    return SizedBox.expand(
      child: DraggableScrollableSheet(
        initialChildSize: sheetPosition,
        builder: (BuildContext context, ScrollController scrollController) {
          return ColoredBox(
            color: colorScheme.primary,
            child: Column(
              children: <Widget>[
                // Drag handle
                GestureDetector(
                  onVerticalDragUpdate: (DragUpdateDetails details) {
                    setState(() {
                      sheetPosition -= details.delta.dy / dragSensitivity;
                      if (sheetPosition < 0.25) {
                        sheetPosition = 0.25;
                      }
                      if (sheetPosition > 1.0) {
                        sheetPosition = 1.0;
                      }
                    });
                  },
                  child: Container(
                    width: double.infinity,
                    color: Colors.transparent,
                    child: Align(
                      alignment: Alignment.topCenter,
                      child: Container(
                        margin: const EdgeInsets.symmetric(vertical: 8.0),
                        width: 32.0,
                        height: 4.0,
                        decoration: BoxDecoration(
                          color: colorScheme.surfaceVariant,
                          borderRadius: BorderRadius.circular(8.0),
                        ),
                      ),
                    ),
                  ),
                ),
                // Scrollable sheet content
                Flexible(
                  child: ListView.builder(
                    controller: scrollController,
                    itemCount: 25,
                    itemBuilder: (BuildContext context, int index) {
                      return ListTile(
                        title: Text(
                          'Item $index',
                          style: TextStyle(color: colorScheme.surface),
                        ),
                      );
                    },
                  ),
                ),
              ],
            ),
          );
        },
      ),
    );
  }
}
```
</details>

<details>
<summary>Demo video of issue</summary>

https://github.com/flutter/flutter/assets/104349824/71c6b43a-67d6-4829-9ebb-d8bc8aa24209

</details>

Reply:
Interesting. I wonder if we should not consider adding the drag handle as a feature provided by the DraggableScrollableSheet? Then we could more easily control the scroll position and sheet height from within since DraggableScrollableSheet has a custom ScrollPosition, _DraggableScrollableSheetScrollPosition

Reply:
Hi Piinks :), I managed to fix this as the code I pushed. Currently lacking tests but I would like to confirm the implementation before writing them.

I tested the initial code with mouse and trackpad, the results I shared at https://github.com/flutter/flutter/issues/101903#issuecomment-1771507814. Accordingly, the behavior on the trackpad is not consistent with the mouse (with default scroll behavior because we will not implement custom scroll behavior in this PR).

To solve the inconsistency and keep the API example not being affected in case #101903 (or other related issues if any) is landed, I propose implementing this PR as follows:

- On Web and desktop platforms: users can only drag the handle/grabber to expand or collapse the sheet. Scroll and drag gestures from the mouse or trackpad will not be able to expand/collapse the sheet (the result will be like the demo video below)
- On mobile platforms, users can still drag the sheet to expand or collapse as it is (handle/grabber will not be displayed).

<details>
<summary>Demo sample code on desktop and web</summary>

| macOS | Web |
| --------------- | --------------- |
<video src="https://github.com/flutter/flutter/assets/104349824/89763abf-8696-4f6b-b42e-cbed0ff532b5"/> | <video src="https://github.com/flutter/flutter/assets/104349824/0b99fed1-d910-49b3-a753-5f6096ad2399"/>

</details>

Please let me know your thoughts about this :)

Reply:
Oh exciting! Thanks for the update! I will check it out!

Reply:
Apologies for the delay, I was out for a few days.

> To solve the inconsistency and keep the API example not being affected in case https://github.com/flutter/flutter/issues/101903 (or other related issues if any) is landed, I propose implementing this PR as follows:
>
> * On Web and desktop platforms: users can only drag the handle/grabber to expand or collapse the sheet. Scroll and drag gestures from the mouse or trackpad will not be able to expand/collapse the sheet (the result will be like the demo video below)
> * On mobile platforms, users can still drag the sheet to expand or collapse as it is (handle/grabber will not be displayed).

I think this makes sense! It is common to have different code paths for slightly different interaction behaviors based on the platform.